apps/simple_history.py

```diff
@@ -0,0 +1,91 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+
+from django.core.serializers import serialize
+from django.db import models
+
+from simple_history.models import HistoricalRecords, ModelChange, ModelDelta
+
+
+RELATED_TO_LOCATION = ('shipment_to', 'shipment_from', 'shipment_dest', 'shipment_bill', )
+
+
+def get_user(request=None, instance=None):
+
+    if request:
+        return request.user.id
+
+    if instance:
+        if hasattr(instance, 'updated_by'):
+            return instance.updated_by
+        for related in RELATED_TO_LOCATION:
+            if hasattr(instance, related):
+                return getattr(instance, related).updated_by
+
+    return None
+
+
+def _model_to_dict(model):
+    return json.loads(serialize("json", [model]))[0]["fields"]
+
+
+class TxmHistoricalRecords(HistoricalRecords):
+
+    def _get_history_user_fields(self):
+        if self.user_id_field:
+            # We simply track the updated_by field
+            history_user_fields = {
+                "history_user": models.CharField(null=True, blank=True, max_length=36),
+            }
+        else:
+            history_user_fields = {}
+
+        return history_user_fields
+
+
+class TxmHistoricalChanges:
```

Comment:
In `django-simple-history`, `HistoricalChanges` is a mixin. I think it makes sense to keep it a mixin in our codebase as well. Remove the constructor and add the mixin to `Shipment` and `Location`. Then change all `self.obj` to `self.instance` and you should be able to do `shipment.diff_against(historical_shipment)` :+1: